sdk/identity/src/device_code_flow/mod.rs

```diff
@@ -184,3 +183,36 @@ async fn post_form(
     req.set_body(form_body);
     http_client.execute_request(&req).await
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use async_trait::async_trait;
+    use std::error::Error;
+
+    #[async_trait]
```

Comment:
Is all of this meant to test that the future from `start` is `Send`? I'd rather do that by using a function that constrains some argument [like in this playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c4f87e3bd5d2b746272feffa5222b0a0).

Reply:
Thanks for looking at this so speedily. Yes, that was a distilled down version of what caused the compiler to shout at me. Great suggestion to use a function with constrained argument, that makes it much more understandable what this is ensuring. Pushed a new commit with the change.